### PR TITLE
Filter visible specifications

### PR DESCRIPTION
**Pull request by klzns**
#### What problem is this solving?

(**Tip for reviewers**: review commit by commit)

Remove specifications that have "Show Specification" not checked

#### How should this be manually tested?

Workspace | Production
---|---
[Link](https://brenovtex--marinbrasil.myvtex.com/cuba-de-embutir-tramontina-ceres-40-bl-em-aco-inox-acetinado-40-x-35-x-19-cm/p) | [Link](https://marinbrasil.myvtex.com/cuba-de-embutir-tramontina-ceres-40-bl-em-aco-inox-acetinado-40-x-35-x-19-cm/p)
image | image

API using the query string `?compSpecs=true`:
https://marinbrasil.vtexcommercestable.com.br/api/catalog_system/pub/products/search/cuba-de-embutir-tramontina-ceres-40-bl-em-aco-inox-acetinado-40-x-35-x-19-cm/p?&compSpecs=true
image

### CODE REVIEW

**Review by iaronaraujo — APPROVED** [review]

**Review by hiagolcm — APPROVED** [review]

File: node/resolvers/search/index.ts [hunk 1]
```diff
@@ -464,7 +464,7 @@ export const queries = {
         products = await search.productsByReference(values, salesChannel)
         break
       case 'sku':
-        products = await search.productBySku(values, salesChannel)
+        products = await search.productsBySku(values, salesChannel)
```
Comment (victorhmp) [inline node/resolvers/search/index.ts:467]: The returned objects from `productBySku` and `productsBySku` are different, aren't they? Could this be an issue?

File: node/clients/search.ts [hunk 2]
```diff
@@ -74,65 +82,84 @@ export class Search extends AppClient {
 
   public product = (slug: string) =>
     this.get<SearchProduct[]>(
-      `/pub/products/search/${this.searchEncodeURI(slug && slug.toLowerCase())}/p`,
+      this.addCompleteSpecifications(`/pub/products/search/${this.searchEncodeURI(slug && slug.toLowerCase())}/p`),
```
Comment (kaisermann) [inline node/clients/search.ts:85]: I know this is not related to this PR, but do we need that short circuiting? `slug &&`? 
```suggestion
      this.addCompleteSpecifications(`/pub/products/search/${this.searchEncodeURI(slug.toLowerCase())}/p`),
```

**Review by victorhmp — APPROVED** [review]

File: node/resolvers/search/product.ts [hunk 3]
```diff
@@ -244,23 +244,38 @@ export const resolvers = {
...
+      const visibleSpecifications = product.completeSpecifications
+        ? product.completeSpecifications.reduce<Record<string, boolean>>((acc, specification) => {
+            acc[specification.Name] = specification.IsOnProductDetails
+            return acc
+          }, {})
+        : {}
+
+      let noTranslationSpecificationGroups = allSpecificationsGroups.map(
+        (groupName: string) => {
+          let groupSpecifications = (product as unknown as DynamicKey<string[]>)?.[groupName] ?? []
```
Comment (kaisermann) [inline node/resolvers/search/product.ts:257]: dios mio esta linha

**Review by kaisermann — APPROVED** [review]

File: node/resolvers/search/index.ts (hunk 1 above)
Comment (klzns) [reply node/resolvers/search/index.ts:467]: They are the same return value, it's an array just like the other methods:

```
products = await search.productByReference(value)
```

File: node/clients/search.ts (hunk 2 above)
Comment (klzns) [reply node/clients/search.ts:85]: I sincerely don't know, I'll leave it be

**Pull request merged**